src/core/qgsvectorlayerutils.h

```diff
@@ -167,6 +167,56 @@ class CORE_EXPORT QgsVectorLayerUtils
      * \since QGIS 3.4
      */
     static std::unique_ptr<QgsVectorLayerFeatureSource> getFeatureSource( QPointer<QgsVectorLayer> layer ) SIP_SKIP;
+
+    /**
+     * Matches the attributes in \a feature to the specified \a fields.
+     *
+     * This causes the attributes contained within the given \a feature to be rearranged (or in
+     * some cases dropped) in order to match the fields and order indicated by \a fields.
```

Comment:
Should there be a note included about the behavior difference if `feature.fields().empty()` vs. a feature with fields set?

Reply:
Fixed in #7998